plugins/woocommerce-blocks/assets/js/settings/shared/default-fields.ts

```diff
@@ -2,12 +2,27 @@
  * External dependencies
  */
 import { ComboboxControlOption } from '@woocommerce/base-components/combobox';
+import type { AllHTMLAttributes, AriaAttributes } from 'react';
 
 /**
  * Internal dependencies
  */
 import { getSetting } from './utils';
 
+// A list of attributes that can be added to a custom field when registering it.
+type CustomFieldAttributes = Pick<
+	AllHTMLAttributes< HTMLInputElement >,
+	| 'maxLength'
+	| 'disabled'
+	| 'readOnly'
+	| 'pattern'
+	| 'placeholder'
```

Comment:
It doesn't make sense to support placeholder since our design system uses the label as a placeholder.

Reply:
OK, I was on the fence about it, it does technically work, and displays fine because of my changes but it would introduce an inconsistency if some people used it and some didn't. Fine to remove it then 👌🏼